src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -17,8 +17,11 @@ namespace Microsoft.AspNetCore.Razor.TagHelpers
     [DebuggerDisplay("{DebuggerToString(),nq}")]
     public class DefaultTagHelperContent : TagHelperContent
     {
-        private List<object> _buffer;
+        private object _singleContent;
+        private bool _isSingleContentSet;
         private bool _isModified;
+        private bool _hasContent;
```

Comment:
Like the new field. But why does the `Buffer` setter still muck with `_isModified` now. Better to manipulate these two fields in fewer places, in particular set `_isModified` to `true` only in `Clear()` and `AppendCore()`.